Firestore/Source/Public/FirebaseFirestore/FIRQuery.h

```diff
@@ -109,6 +109,30 @@ NS_SWIFT_NAME(Query)
 - (FIRQuery *)queryWhereField:(NSString *)field
                     isEqualTo:(id)value NS_SWIFT_NAME(whereField(_:isEqualTo:));
 
+/**
+ * Creates and returns a new `FIRQuery` with the additional filter that documents must
+ * contain the specified field and the value does not equal the specified value.
+ *
+ * @param path The path of the field to compare.
+ * @param value The value the field must be equal to.
+ *
+ * @return The created `FIRQuery`.
+ */
+- (FIRQuery *)queryWhereFieldPath:(FIRFieldPath *)path
```

Comment:
Also, we can't make these public API changes until we're ready to launch. Are you going to hold this PR indefinitely?

In the past we've made these part of the Internal Query category to be able to commit the bulk of the code and tests and just delay the public release.